core/src/main/java/org/elasticsearch/index/query/DisMaxQueryBuilder.java

```diff
@@ -54,6 +61,24 @@ public DisMaxQueryBuilder add(QueryBuilder queryBuilder) {
     }
 
     /**
+     * Add a list of sub-queries to this disjunction.
+     */
+    public DisMaxQueryBuilder add(List<QueryBuilder> queryBuilders) {
+        for (QueryBuilder<?> queryBuilder : queryBuilders) {
+            add(queryBuilder);
+        }
+        return this;
+    }
+
+
+    /**
+     * @return an immutable list copy of the current sub-queries of this disjunction
+     */
+    public List<QueryBuilder> queries() {
+        return ImmutableList.copyOf(this.queries);
```

Comment:
I dunno, I think we should do the same in all queries though, whatever we decide to do. we have a public method to add queries, so this list can be modified from the outside, why can't we just return the same list through getter? What is the advantage of making a copy here?